bindings/node/lib/stateMachine.js

```diff
@@ -314,8 +315,7 @@ module.exports = function(modules) {
           buffer.append(data);
           while (request.bytesNeeded > 0 && buffer.length) {
             const bytesNeeded = Math.min(request.bytesNeeded, buffer.length);
-            request.addResponse(buffer.slice(0, bytesNeeded));
-            buffer.consume(bytesNeeded);
```

Comment:
Our `read` method will accomplish the same thing in one call, returning a new copy of the bytes and dropping them from the underlying bufferpool.